chat-widget/src/components/prechatsurveypanestateful/PreChatSurveyPaneStateful.tsx

```diff
@@ -60,15 +61,24 @@ export const PreChatSurveyPaneStateful = (props: IPreChatSurveyPaneStatefulParam
         onSubmit: async (values: { index: number, label: any, id: any, value: string }[]) => {
             TelemetryHelper.logActionEvent(LogLevel.INFO, { Event: TelemetryEvent.PrechatSubmitted });
             dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Loading });
+
             try {
-                const prechatResponseValues = extractPreChatSurveyResponseValues(state.domainStates.preChatSurveyResponse, values);
-                const optionalParams = {
-                    initContext: {
-                        preChatResponse: prechatResponseValues
-                    }
-                };
-                setPreChatResponseEmail(values);
-                await initStartChat(optionalParams);
+                const widgetStateFromCache = DataStoreManager.clientDataStore?.getData(Constants.widgetStateDataKey, "localStorage");
+                const persistedState = widgetStateFromCache ? JSON.parse(widgetStateFromCache) : undefined;
+                let optionalParams = {};
+                if (persistedState?.domainStates?.chatToken) {
+                    optionalParams = { liveChatContext: { chatToken: persistedState?.domainStates?.chatToken } };
```

Comment:
Should we just simply use `liveChatContext` instead of building a custom `liveChatContext`?